docs/recipes/atom-with-listeners.mdx

```diff
@@ -0,0 +1,81 @@
+---
+title: Atom creator with listeners
+nav: 8.07
+keywords: creators,listeners
+---
+
+## atomWithListeners
+
+> `atomWithListeners` creates an atom and a hook. The hook can be called to
+> add a new listener. The hook takes as an argument a callback, and that
+> callback is called every time the atom's value is set. The hook also
+> returns a function to remove the listener.
+
+This can be useful when you want to create a component that can listen to when
+an atom's state changes without having to re-render that component with each of
+those state changes.
+
+```ts
+import { useEffect } from 'react'
+import { atom, useSetAtom, Getter, Setter, SetStateAction } from 'jotai'
+
+type Callback<Value> = (
+  get: Getter,
+  set: Setter,
+  newVal: Value,
+  prevVal: Value
+) => void
+
+export function atomWithListeners<Value>(initialValue: Value) {
```

Comment:
I think `useCurrentStore` (non-exist API) and `store.sub(aAtom)` would be better

Reply:
This recipe is written for v1 API, I guess. `useStore` exists, btw.